### PR TITLE
mailing through sendgrid

### PR DESCRIPTION
**Pull request by purushottam08**

### CODE REVIEW

File: kafka-consumers/notification-consumer/pom.xml [hunk 1]
```diff
@@ -77,7 +77,11 @@
 			<groupId>javax.validation</groupId>
 			<artifactId>validation-api</artifactId>
 		</dependency>
-		
+
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-web</artifactId>
+		</dependency>
```
Comment (shunaray) [inline kafka-consumers/notification-consumer/pom.xml:84]: update the version of future compatibility risk

File: kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/config/SendGridEmailConsumerConfig.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.publicissapient.kpidashboard.notification.config;
```
Comment (shunaray) [inline kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/config/SendGridEmailConsumerConfig.java:1]: copyright missing

File: kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/model/EmailTemplate.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package com.publicissapient.kpidashboard.notification.model;
```
Comment (shunaray) [inline kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/model/EmailTemplate.java:1]: cpyright

File: kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/service/NotificationServiceImpl.java [hunk 4]
```diff
@@ -1,96 +1,147 @@
 package com.publicissapient.kpidashboard.notification.service;
```
Comment (shunaray) [inline kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/service/NotificationServiceImpl.java:1]: include cpyright in all files where missing

File: kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/model/EmailTemplate.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
```
Comment (shunaray) [inline kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/model/EmailTemplate.java:13]: add @ author in new file created

File: kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/service/NotificationService.java [hunk 6]
```diff
@@ -3,7 +3,9 @@
 import com.publicissapient.kpidashboard.notification.model.EmailEvent;
 
 public interface NotificationService {
-	
-	public void sendMail(String key,EmailEvent emailEvent);
+
+    void sendMail(String key, EmailEvent emailEvent);
```
Comment (shunaray) [inline kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/service/NotificationService.java:7]: add javadoc

File: kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/service/NotificationServiceImpl.java [hunk 7]
```diff
@@ -1,96 +1,147 @@
...
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return headers;
+    }
+
+    public void sendMail(String key, EmailEvent emailEvent) {
+        try {
+            Context context = createContext(emailEvent.getCustomData());
+            String html = processTemplate(key, context);
+            sendEmailViaJMS(emailEvent, html);
+        } catch (Exception e) {
```
Comment (shunaray) [inline kafka-consumers/notification-consumer/src/main/java/com/publicissapient/kpidashboard/notification/service/NotificationServiceImpl.java:58]: avoid generic exception catch, use more acurate catch block

**Review by shunaray — APPROVED** [review]

**Review by kunkambl — APPROVED** [review]

**Pull request merged**